### ISSUE TITLE
Possible bug in BlenderDNA.inl

### ISSUE THREAD

**Issue by Wasabi375** [assimp/assimp#2160]
Hey, I'm one of the guys trying to port [assimp to Kotlin](https://github.com/kotlin-graphics/assimp) and I found this strange part in [BlenderDNA.inl ll. 518-527](https://github.com/assimp/assimp/blob/master/code/BlenderDNA.inl#L518-L527)

```c++
bool Structure :: ResolvePointer(...) {
...
bool res = false;
// allocate raw storage for the array
out.resize(num);
for (size_t i = 0; i< num; ++i) {
    Pointer val;
    Convert(val,db);

    // and resolve the pointees
    res = ResolvePointer(out[i],val,db,f) && res;
}
...
return res;
```
Shouldn't this be either `res = ResolvePointer(out[i], val, db, f) || res;` or `bool res = true;`?
Based on what I can tell `ResolvePointer` should return true if the data was already cached. I'm not quite sure whether this means all the data or just some of it. If it should return true if all of it was already cached `res` should start as `true` and it needs to be `&&` otherwise it would need to be `res = false` and `||` or am I totally mistaken here?

I don't think this leads to any errors, because as far as I can tell the return value is actually never used for this branch, but I might be mistaken.

**Comment by kimkulling** [comment]
You are right, this looks spooky. By the way: nice work, the Kotlin port. It would be great if you could prepare a pull-request to add a link to your work.

**Comment by tellypresence** [comment]
Any updates on this?

**Comment by tellypresence** [comment]
Closing due to no response
